src/routes/catalog.ts

```diff
@@ -133,9 +142,9 @@ export const providerCatalogHandler = async (req: Request, res: Response) => {
 
   const selfLinks = generateSelfLinks(req, cursor, collections?.length);
 
-  const childLinks = (collections ?? []).map(({ id, title }) => ({
+  const childLinks = (collections ?? []).map(({ id, title, provider }) => ({
     rel: "child",
-    href: `${getBaseUrl(self)}/collections/${encodeURIComponent(id)}`,
+    href: `${getBaseUrl(self).replace("ALL", provider)}/collections/${encodeURIComponent(id)}`,
```

Comment:
If a provider happens to have "ALL" in their name, this could mess up the link.
EX: LPFALL link would be changed to LPFLPFALL.
But not sure if any providers have ALL in their name or could have it in the future.

Reply:
Good spot. In other places I've used '/ALL/' to avoid that problem. I will update.

Reply:
Updated.